### PR TITLE
Nested attribute docs improvements

### PR DESCRIPTION
**Pull request by ghiculescu**
2 small improvemennts to the docs for `ActiveRecord::NestedAttributes`.

- Include a link to `ActionView::Helpers::FormHelper::fields_for`, since you would typically use these tools in tandem I think it makes sense to point you to it.
- Show an example of how to write an integration test when using `ActionView::Helpers::FormHelper::fields_for`. You could argue that this example should also go in Action View, but I think it makes more sense here as the "parent" doc for the Nested Attributes feature.

### CODE REVIEW

**Comment by ghiculescu** [comment]
The red build is unrelated (this is docs only).

File: activerecord/lib/active_record/nested_attributes.rb [hunk 1]
```diff
@@ -280,6 +280,26 @@ class TooManyRecords < ActiveRecordError
     #   member = Member.new
     #   member.avatar_attributes = {icon: 'sad'}
     #   member.avatar.width # => 200
+    #
+    # === Forms
+    #
+    # Use ActionView::Helpers::FormHelper::fields_for to create form elements
```
Comment (skipkayhil) [inline activerecord/lib/active_record/nested_attributes.rb:286]:
```suggestion
    # Use ActionView::Helpers::FormHelper#fields_for to create form elements
```

File: activerecord/lib/active_record/nested_attributes.rb [hunk 2]
```diff
@@ -280,6 +280,26 @@ class TooManyRecords < ActiveRecordError
...
+    # for updating or destroying nested attributes.
+    #
+    # === Testing
+    #
+    # If you are using ActionView::Helpers::FormHelper::fields_for, your integration
```
Comment (skipkayhil) [inline activerecord/lib/active_record/nested_attributes.rb:291]:
```suggestion
    # If you are using ActionView::Helpers::FormHelper#fields_for, your integration
```

File: activerecord/lib/active_record/nested_attributes.rb [hunk 3]
```diff
@@ -280,6 +280,26 @@ class TooManyRecords < ActiveRecordError
...
+    # tests should replicate the HTML structure it provides. For example;
+    #
+    #   post members_path, params: {
+    #     member: {
+    #       name: 'joe',
+    #       posts_attributes: {
+    #         "0" => { title: 'Foo' },
+    #         "1" => { title: 'Bar' }
```
Comment (skipkayhil) [inline activerecord/lib/active_record/nested_attributes.rb:299]:
```suggestion
    #       name: "joe",
    #       posts_attributes: {
    #         "0" => { title: "Foo" },
    #         "1" => { title: "Bar" }
```

**Review by skipkayhil — COMMENTED** [review]
A few small formatting changes

File: activerecord/lib/active_record/nested_attributes.rb (hunk 3 above)
Comment (ghiculescu) [reply activerecord/lib/active_record/nested_attributes.rb:299]: I prefer double quotes, but the rest of this doc uses single quotes. See https://api.rubyonrails.org/classes/ActiveRecord/NestedAttributes/ClassMethods.html

Comment (skipkayhil) [reply activerecord/lib/active_record/nested_attributes.rb:299]: Good point, I was mostly interested in consistency (either the ones I suggested or the `"0"` and `"1"`)

Comment (ghiculescu) [reply activerecord/lib/active_record/nested_attributes.rb:299]: Oh yeah of course... I'll fix the 0 and 1.

**Review by skipkayhil — APPROVED** [review]

File: activerecord/lib/active_record/nested_attributes.rb [hunk 4]
```diff
@@ -280,6 +280,26 @@ class TooManyRecords < ActiveRecordError
...
+    #
+    # === Forms
+    #
+    # Use ActionView::Helpers::FormHelper#fields_for to create form elements
+    # for updating or destroying nested attributes.
+    #
+    # === Testing
+    #
+    # If you are using ActionView::Helpers::FormHelper#fields_for, your integration
+    # tests should replicate the HTML structure it provides. For example;
```
Comment (p8) [inline activerecord/lib/active_record/nested_attributes.rb:292]: Doesn't this always apply when using `accepts_nested_attributes_for`?
Even when you're not using `fields_for`?

File: activerecord/lib/active_record/nested_attributes.rb [hunk 5]
```diff
@@ -280,6 +280,26 @@ class TooManyRecords < ActiveRecordError
     #   member = Member.new
     #   member.avatar_attributes = {icon: 'sad'}
     #   member.avatar.width # => 200
+    #
+    # === Forms
```
Comment (p8) [inline activerecord/lib/active_record/nested_attributes.rb:284]: Maybe expand the title a bit?
```suggestion
    # === Creating forms with nested attributes
```

File: activerecord/lib/active_record/nested_attributes.rb (hunk 4 above)
Comment (p8) [reply activerecord/lib/active_record/nested_attributes.rb:292]: I'm also wondering if describing the integration with ActionView and ActionController is more suited for the guides.

Comment (ghiculescu) [reply activerecord/lib/active_record/nested_attributes.rb:292]: @p8 i agree with these suggestions but I won’t get to them for a little while. If you like, I’d love for you to take over the PR.

Comment (p8) [reply activerecord/lib/active_record/nested_attributes.rb:292]: @ghiculescu Sure! 😄

**Pull request merged**